editor/src/components/navigator/navigator.spec.browser2.tsx

```diff
@@ -3855,3 +3864,522 @@ describe('Navigator labels', () => {
+              </Group>
+            </Group>
+          </div>
+        `),
+      )
+    })
+    it('adjusts the groups when reparenting out of a nested group', async () => {
+      const renderResult = await renderTestEditorWithCode(
+        makeTestProjectCodeWithSnippet(`
+          <div data-uid='root'>
+            <Group
+              data-uid='group-1'
+              style={{
+                background: 'white',
+                position: 'absolute',
+                left: 10,
+                top: 10,
+                width: 390,
+                height: 390,
+              }}
+            >
+              <div data-uid='intruder' style={{
+                background: 'orange',
+                width: 20,
+                height: 20,
+                position: 'absolute',
+                left: 100,
+                top: 100,
+              }} />
+              <Group
+                data-uid='group-2'
+                style={{
+                  background: 'cyan',
+                  position: 'absolute',
+                  left: 0,
+                  top: 0,
+                  width: 390,
+                  height: 390,
+                }}
+              >
+                <div data-uid='dragme' style={{
+                  background: 'red',
+                  width: 50,
+                  height: 50,
+                  position: 'absolute',
+                  left: 0,
+                  top: 0
+                }} />
+                <div data-uid='group-child-1' style={{
+                  background: 'gray',
+                  width: 50,
+                  height: 50,
+                  position: 'absolute',
+                  left: 150,
+                  top: 150
+                }} />
+                <div data-uid='group-child-2' style={{
+                  background: 'gray',
+                  width: 50,
+                  height: 50,
+                  position: 'absolute',
+                  left: 340,
+                  top: 340
+                }} />
+              </Group>
+            </Group>
+          </div>
+      `),
+        'await-first-dom-report',
+      )
+
+      const dragmePath = EP.fromString(
+        `${BakedInStoryboardUID}/${TestSceneUID}/${TestAppUID}:root/group-1/group-2/dragme`,
+      )
+
+      const rootPath = EP.fromString(`${BakedInStoryboardUID}/${TestSceneUID}/${TestAppUID}:root`)
+
+      const canvasRect = renderResult.renderedDOM.getByTestId('canvas-root').getBoundingClientRect()
+      const canvasCenter = getRectCenter(
+        windowRectangle({
+          x: canvasRect.x + DefaultNavigatorWidth,
+          y: canvasRect.y,
+          width: canvasRect.width - DefaultNavigatorWidth,
+          height: canvasRect.height,
+        }),
+      )
+
+      await renderResult.dispatch(
+        [reorderComponents([dragmePath], rootPath, front(), canvasPoint(canvasCenter))],
+        true,
+      )
+
+      expect(getPrintedUiJsCode(renderResult.getEditorState())).toEqual(
```

Comment:
one idea: you could also assert that the globalFrame of the reparented element stays the same (as that is the goal of the behavior)

Reply:
makes sense!

Reply:
https://github.com/concrete-utopia/utopia/pull/3936/commits/53596164aaaaed31e3743832a2ab0e3c21583825